src/ext/java/org/opentripplanner/ext/flex/trip/UnscheduledTrip.java

```diff
@@ -64,9 +69,10 @@ public UnscheduledTrip(UnscheduledTripBuilder builder) {
 
     for (int i = 0; i < size; i++) {
       this.stopTimes[i] = new StopTimeWindow(stopTimes.get(i));
-      this.dropOffBookingInfos[i] = stopTimes.get(0).getDropOffBookingInfo();
-      this.pickupBookingInfos[i] = stopTimes.get(0).getPickupBookingInfo();
+      this.dropOffBookingInfos[i] = stopTimes.get(i).getDropOffBookingInfo();
+      this.pickupBookingInfos[i] = stopTimes.get(i).getPickupBookingInfo();
```

Comment:
Are you fixing some unrelated bug here?

Reply:
Yes, there was a bug when you had several zones with individual booking rules then only the first one would be used for all zones.